enhancements/monitoring/hypershift-monitoring.md

```diff
@@ -0,0 +1,392 @@
+
+> NOTE: The definition of “monitor” in the bullet list below means ability to graph, query and alert on near-real-time metrics about involved components.
+
+* HSRE can monitor management clusters and control planes running there.
+  * Ability to use AlertManager and PagerDuty workflow
+  * Ability to monitor Tugboats themselves
+  * Ability to monitor Customer control planes
+  * Ability to view fleet-level metrics and drill-downs in a single pane of glass
+  * Ability to keep metrics in-region
+* HSRE can monitor control-plane related (managed) software on guest clusters:
+  * For example Ingress/Router/CNI/CSI 
+  
+> Exact list of what is scraped on guest clusters for HSRE does not matter. All we need to know: some service monitors.
+
+* HSRE can query metrics data over SLO period (usually 28 days).
+  * Longer durations nice to have, but technically not required. There is also option that allows HSRE to persist SLO from platform to [different backend if needed](https://docs.google.com/document/d/1BJarERppgiJ8esc6d8anbJMOQ0AflFBQea-Zc9wAp0s/edit?disco=AAAAQTFsbwM).
+* HSRE have an easy way to monitor dozens of HyperShift clusters in multiple geo regions and thousands of control planes in total. This means in particular:
+  * Ability to perform PromQL across fleet metrics
+
+> NOTE(Mariusz): Alerts can be region wide only, and query in different tab per region might be good enough for now.
+
+  * Ability to see logs across the fleet to see a pattern
+  * No need to tunnel to specific control plane to query metrics or see logs
+  * Centralized alerting
+* CUS can monitor / use HPA/VPA / vis with local console using metrics FROM their workloads on guest clusters or from nodes on guest clusters
+  * Container metrics (about their containers).
+  * This includes part of Kube State Metrics (Kubernetes metrics) that relate to their workloads.
+* BU can gather required telemetry using the Telemeter pipeline.
+
+### Non-Goals
+
+* Solve Logging and Tracing needs.
+* Provide scalable storage solution for customer metrics. We just want to ensure UWM, their DIY,
+console or vendor monitoring does not have major blockers when integrating with this
+architecture.
+* Describe Managed Services (Addons) monitoring in detail. We assume Managed Services are just “Customer workloads”, where the MTSRE is just a customer. See [What About Addons FAQ](#what-about-layered-service--addons-monitoring) for more details.
+* Solve observability challenges brought by KCP. KCP has different trade-offs.
+* Provide any other UI than Grafana for our Managed HSRE needs.
+* Provide any other querying APIs than Prometheus based on PromQL
+* Give customers access to any control plane metrics unrelated to their workloads (this is similar in GKE).
+* Provide a solution for non-HyperShift OSD. In fact, the solution for OSD observability is planned to be really similar to HyperShift. 
```

Comment:
Some convergence here might be logical.  We can see how it goes though, no rush.  A decision by end of Internal Preview phase would be desirable.

Reply:
Hm, I think you statement is a bit conflicting here:

You mentioned:

>  leave any and all requirements for a central metrics aggregation solution out

and

>  IMHO from the POV of hypershift it is a blackbox service the hypershift components shouldn't need to know anything about except how to communicate with it.

But in order to allow HyperShift to treat it as black-box we need to change architecture to centralized approach. That is what simplifies client (from perspective of observability) workload. 

Still, while NON-HyperShift clusters are out of the discussion for this enhancement, I believe aiming for both short term (enabling Hypershift) as well as long term (all clusters have uniform observability) with one go is pretty tempting here! (:

Reply:
Let's provide EPIC for it.